src/node/txreconciliation.cpp

```diff
@@ -142,9 +250,104 @@ class TxReconciliationTracker::Impl
+                best_peers.emplace_back(hash_key, indexed_state.first);
+            }
+        }
+
+        // Sort by the assigned key.
+        std::sort(best_peers.begin(), best_peers.end());
+        best_peers.resize(targets_size);
+
+        std::vector<NodeId> new_fanout_candidates;
+        new_fanout_candidates.reserve(targets_size);
+        for_each(best_peers.begin(), best_peers.end(),
+                 [&new_fanout_candidates](auto& keyed_peer) { new_fanout_candidates.push_back(keyed_peer.second); });
+
+        // Sort by NodeId.
+        std::sort(new_fanout_candidates.begin(), new_fanout_candidates.end());
+        bool found = std::binary_search(new_fanout_candidates.begin(), new_fanout_candidates.end(), peer_id);
+
+        // If the cache is full, make room for the new entry.
+        if (m_tx_fanout_targets_cache_order.size() == FANOUT_TARGETS_PER_TX_CACHE_SIZE) {
+            auto expired_tx = m_tx_fanout_targets_cache_order.front();
+            m_tx_fanout_targets_cache_data.erase(expired_tx);
+            m_tx_fanout_targets_cache_order.pop_front();
+        }
+        m_tx_fanout_targets_cache_data.emplace(wtxid, std::move(new_fanout_candidates));
+        m_tx_fanout_targets_cache_order.push_back(wtxid);
+
+        return found;
+    }
+
+    bool ShouldFanoutTo(const Wtxid& wtxid, NodeId peer_id,
+                        size_t inbounds_nonrcncl_tx_relay, size_t outbounds_nonrcncl_tx_relay)
+        EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        auto peer_state = GetRegisteredPeerState(peer_id);
+        if (!peer_state) return true;
+        // We decide whether a particular peer is a low-fanout flood target differently
+        // based on its connection direction:
+        // - for outbounds we have a fixed number of flood destinations;
+        // - for inbounds we use a fraction of all inbound peers supporting tx relay.
+        //
+        // We first decide how many reconciling peers of a given direction we want to flood to,
+        // and then generate a list of peers of that size for a given transaction. We then see
+        // whether the given peer falls into this list.
+        double destinations;
+        if (peer_state->m_we_initiate) {
+            destinations = OUTBOUND_FANOUT_DESTINATIONS - outbounds_nonrcncl_tx_relay;
+        } else {
+            // Since we use the fraction for inbound peers, we first need to compute the total
+            // number of inbound targets.
+            const double inbound_targets = (inbounds_nonrcncl_tx_relay + m_inbounds_count) * INBOUND_FANOUT_DESTINATIONS_FRACTION;
+            destinations = inbound_targets - inbounds_nonrcncl_tx_relay;
+        }
+
+        // Pure optimization to avoid going through the peers when the odds of picking one are
+        // too low.
+        if (destinations < 0.001) {
```

Comment:
can be constified same than `OUTBOUND_FANOUT_DESTINATIONS` and `INBOUND_FANOUT_DESTINATIONS_FRACTION` as explanation for this value is tight to them

Reply:
Not really related. Change these constants in any way, and the `0.001` value won't be affected.

Reply:
okay will read back the txrelayism issue on const selection.